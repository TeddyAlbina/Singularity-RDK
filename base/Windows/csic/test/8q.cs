class EightQueens {
	static bool[]
		up =  { true, true, true, true, true, true, true, true, true, true, true, true, true, true, true },
		down = (bool[])up.Clone(),
		rows = new bool[8] { true, true, true, true, true, true, true, true };
	static public void Main() {
		for (int i = 0; i < up.Length; i++)
			up[i] = down[i] = true;
		for (int i = 0; i < rows.Length; ++i)
			rows[i] = true;
		queens(0, new int[8]);
	}
	static void queens(int c, int[] x) {
		for (int r = 0; r < rows.Length; r++)
			if (rows[r] && up[r-c+7] && down[r+c]) {
				rows[r] = up[r-c+7] = down[r+c] = false;
				x[c] = r;
				if (c == 7)
					print(x);
				else
					queens(c + 1, x);
				rows[r] = up[r-c+7] = down[r+c] = true;
			}
	}
	static void print(int[] x) {
		foreach (int c in x)
			System.Console.Write("{0}", c + 1);
		System.Console.WriteLine();
	}
}
