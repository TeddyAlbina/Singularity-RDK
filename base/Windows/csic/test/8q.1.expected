15863724
16837425
17468253
17582463
24683175
25713864
25741863
26174835
26831475
27368514
27581463
28613574
31758246
35281746
35286471
35714286
35841726
36258174
36271485
36275184
36418572
36428571
36814752
36815724
36824175
37285146
37286415
38471625
41582736
41586372
42586137
42736815
42736851
42751863
42857136
42861357
46152837
46827135
46831752
47185263
47382516
47526138
47531682
48136275
48157263
48531726
51468273
51842736
51863724
52468317
52473861
52617483
52814736
53168247
53172864
53847162
57138642
57142863
57248136
57263148
57263184
57413862
58413627
58417263
61528374
62713584
62714853
63175824
63184275
63185247
63571428
63581427
63724815
63728514
63741825
64158273
64285713
64713528
64718253
68241753
71386425
72418536
72631485
73168524
73825164
74258136
74286135
75316824
82417536
82531746
83162574
84136275
